connect/runtime/src/test/java/org/apache/kafka/connect/integration/MonitorableSourceConnector.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.integration;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.runtime.TestSourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.tools.ThroughputThrottler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+/**
+ * A source connector that is used in Apache Kafka integration tests to verify the behavior of
+ * the Connect framework, but that can be used in other integration tests as a simple connector
+ * that generates records of a fixed structure. The rate of record production can be adjusted
+ * through the configs 'throughput' and 'messages.per.poll'
+ */
+public class MonitorableSourceConnector extends TestSourceConnector {
```

Comment:
couple of questions over this connector:
- since this is a "monitorable" connector, we'd need to be able to monitor something: for example: when the connector or its tasks come up or how many records the tasks source.
- one of the problems we ran into the sink connector was that it would start syncing records as soon as the first task starts. this became problematic when running multi task tests (and we are trying to get all tasks to read at process a fixed number of records, for example). I understand that we are not sourcing records in the tests in this PR, and therefore those hooks are not in place. but for the sake of completion, we should either put those in place, or mention here (or in a ticket) that this test needs extra retrofitting before it can be used in integration tests that move records around.

Reply:
This is a valid point. The integration framework is new, so I'm expecting we'll extending gradually. 
But this is valid point, so I added another commit that uses the handle's in both connector types and also confirms commits.